packages/@aws-cdk/aws-cloudfront/lib/edge-function.ts

```diff
@@ -0,0 +1,209 @@
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as ssm from '@aws-cdk/aws-ssm';
+import {
+  App, BootstraplessSynthesizer, Construct, ConstructNode, DefaultStackSynthesizer, IStackSynthesizer,
+  ResourceEnvironment, Stack, StackProps, Token,
+} from '@aws-cdk/core';
+import * as resources from '@aws-cdk/custom-resources';
+
+/** Properties for creating a Lambda@Edge function */
+export interface EdgeFunctionProps extends lambda.FunctionProps { }
+
+/**
+ * A Lambda@Edge function.
+ *
+ * Convenience construct for requesting a Lambda function in the 'us-east-1' region for use with Lambda@Edge.
+ * Implements several restrictions enforced by Lambda@Edge.
+ */
+export class EdgeFunction extends Construct implements lambda.IVersion {
+
+  private static readonly EDGE_REGION: string = 'us-east-1';
+
+  public readonly edgeArn: string;
+  public readonly functionName: string;
+  public readonly functionArn: string;
+  public readonly grantPrincipal: iam.IPrincipal;
+  public readonly isBoundToVpc = false;
+  public readonly lambda: lambda.IFunction;
+  public readonly permissionsNode: ConstructNode;
+  public readonly role?: iam.IRole;
+  public readonly version: string;
+
+  public readonly stack: Stack;
+  public readonly env: ResourceEnvironment;
+
+  private readonly functionStack: CrossRegionLambdaStack;
+  private readonly currentVersion: lambda.IVersion;
+
+  constructor(scope: Construct, id: string, props: EdgeFunctionProps) {
+    super(scope, id);
+
+    this.stack = Stack.of(this);
+    this.env = {
+      account: this.stack.account,
+      region: this.stack.region,
+    };
+
+    this.functionStack = this.getOrCreateEdgeStack();
+    this.stack.addDependency(this.functionStack);
+
+    const parameterName = `EdgeFunctionArn-${id}`;
+    const { edgeFunction, currentVersion } = this.functionStack.addEdgeFunction(id, parameterName, props);
+    this.currentVersion = currentVersion;
+
+    const parameterArn = this.stack.formatArn({
+      service: 'ssm',
+      region: EdgeFunction.EDGE_REGION,
+      resource: 'parameter',
+      resourceName: parameterName,
+    });
+
+    const customResource = new resources.AwsCustomResource(scope, `${id}FunctionArnReader`, {
+      policy: resources.AwsCustomResourcePolicy.fromStatements([
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ['ssm:GetParameter'],
+          resources: [parameterArn],
+        }),
+      ]),
+      onUpdate: {
+        service: 'SSM',
+        action: 'getParameter',
+        parameters: { Name: parameterName },
+        region: EdgeFunction.EDGE_REGION,
+        // Update physical id to always fetch the latest version
+        physicalResourceId: resources.PhysicalResourceId.of(Date.now().toString()),
+      },
+    });
```

Comment:
This is supported directly by CloudFormation using SSM parameters - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#aws-ssm-parameter-types.

You should be able to use the `StringParameter` construct from the `aws-ssm` module to do this - https://github.com/aws/aws-cdk/blob/e349004a522e2123c1e93bd3402dd7c3f9c5c17c/packages/%40aws-cdk/aws-ssm/lib/parameter.ts#L327

Reply:
Oh wait... does it not work because it's cross-region?

Reply:
I believe so. I'll do an additional test to confirm though. It's possible we can introduce an optional `region` parameter to `StringParameterAttributes` (as we've done elsewhere) to support importing cross-region.